src/tests/index.test.js

```diff
@@ -0,0 +1,549 @@
+describe('Homework - 4', () => {
+    describe('Test init method', () => {
+        describe('ArrayList.init', function (){
+            it('should be defined', function() {
+                assert.isDefined(ArrayList.init, 'method has been defined')
+            });
+            it('should warning user to [] (undefined)', () => {
+                const arr = undefined;
+                const expSize = 'Please input an array';
+                const expArr = 'Please input an array';
+
+                const actual = ArrayList.init(arr);
+
+                assert.deepEqual(actual, expArr);
+                assert.deepEqual(actual, expSize)
+            });
+
+            it('should warning user to [] (null)', () => {
+                const arr = null;
+                const expArr = 'Please input an array';
+                const expSize = 'Please input an array';
+
+                const actual = ArrayList.init(arr);
+
+                assert.deepEqual(actual, expArr);
+                assert.deepEqual(actual, expSize)
+            });
+
+            it('should return [1,2,3]', () => {
+                const arr = [];
+                const arrElm = [1,2,3];
+                const expArr = [1,2,3];
+                const expSize = 3;
+                ArrayList.init(arr);
+
+                const actual = ArrayList.init(arrElm);
+
+                assert.deepEqual(actual, expArr);
+                assert.deepEqual(actual.length, expSize)
+            });
+            
+            it('should return [2]', () => {
+                const arr = [];
+                const arrElm = [2];
+                const expArr = [2];
+                const expSize = 1;
+                ArrayList.init(arr);
+
+                const actual = ArrayList.init(arrElm);
+
+                assert.deepEqual(actual, expArr);
+                assert.deepEqual(actual.length, expSize);
+            });
+
+            it('should return ["Javascript", "is", "awesome"]',() => {
+                const arr = [];
+                const arrElm = ["Javascript", "is", "awesome"];
+                const expArr = ["Javascript", "is", "awesome"];
+                const expSize = 3;
+
+                ArrayList.init(arr);
+
+                const actual = ArrayList.init(arrElm);
+
+                assert.deepEqual(actual, expArr);
+                assert.deepEqual(actual.length, expSize);
+            });
+        });
+    });
+    describe('Test AdAtEnd method', () => {
+        describe('ArrayList.init', function () {
+            it('should be defined', function () {
+                assert.isDefined(ArrayList.addAtEnd, 'method has been defined')
+            });
+
+            it('should warning user to [] (null)', () => {
+                const arr = null;
+                const expArr = 'Please input an array';
+                const expSize = 'Please input an array';
```

Comment:
why the size a string type ? You wrote that the answer should be an empty array but returned a message. If you are returning message please use right name for  variable not expArr .